### DISCUSSION TITLE
Git clone error due to space in file path

### DISCUSSION THREAD

**Discussion by shoooo** [community/community#61503] [Repositories]
### Select Topic Area

Bug

### Body

Hi, 

I am experiencing an issue when cloning a repository where a file has a space in its file path that seems to be inconsistently present. 

I believe that the file path in the repository does have a space, but it somehow decodes into a file path without a space while git cloning and it does not recognize the file, failing and showing 0 files in my local folder as a result. 

This problem occurs on both my desktop and laptop running Windows 10 and 11. Other developers on my team on Windows and Mac systems do not seem to have this issue.

The error message is:
```
error: invalid path 'xxx/xxx/xxx/xxx/xxx /index.ts'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

Here are the steps I've taken to try to resolve the problem:

- Escaped spaces in the file path.
- Enabled long paths by setting git config core.longpaths true.
- Attempted to force checkout with git checkout -f HEAD.
- Changed Git versions (tried both 2.41.0.windows.3 and 2.21.0.windows.3).
- Switched shells (tried Git Bash, PowerShell, and Command Prompt).
- Adjusted settings to allow special characters in file paths that are typically not allowed on Windows systems.
- Checked the commit history, found the file path does contain a space.
- Downloaded the repository as a ZIP file and the file path in question also contains a space.

The space in the file path seems to be inconsistently present. When I manually select and copy the file path, the space does not appear. However, when I use the "copy" button on the GitHub web interface, or clone the repository, or download the ZIP file, the space does appear. Additionally, the space is present in the file path when viewed in the commit history.

I can't explain the issue other than perhaps the file path during the clone operation is somehow different from the actual file path in the repository.

**Comment by Matt-Osborn** [comment]
I know this sounds overly simple, but have you tried double quotes:  " "  instead of the single quotes ' ' that you have?

**Reply by shoooo** [reply]
Hi Matt, 

Thanks for the comment. Do you mean double quotation as in the repository HTTPS like below? 
`git clone "https://github.com/orgname/reponame.git"`

If so, I did put it as a double quotation, and the single quotations are just the error message I got.

**Comment by Matt-Osborn** [comment]
Yes - .That's what I meant. Some interpreters will deal with double and
single quotes differently.

Always worth a shot trying both.

Good luck with your issue!

**Comment by ClassyJuhi** [comment]
1. Use HTTPS instead of SSH for cloning.
2. Update Git to the latest version.
3. Configure Git with git config core.quotePath false.
4. Clone the repository using a different Git client or GUI.

**Reply by shoooo** [reply]
Hi Juhi, sorry I completely missed your message.

I tried all your suggestions and it didn't work so I ended up asking another person to delete the space and commit.. 

I got it working after the space was deleted. It probably was something to do with Windows and Mac environment. 

Thanks for your suggestions🙏
